pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -185,20 +188,71 @@ func (r *ReconcileZookeeperCluster) reconcileStatefulSet(instance *zookeeperv1be
 	} else if err != nil {
 		return err
 	} else {
-		r.log.Info("Updating StatefulSet",
-			"StatefulSet.Namespace", foundSts.Namespace,
-			"StatefulSet.Name", foundSts.Name)
-		zk.SyncStatefulSet(foundSts, sts)
-		err = r.client.Update(context.TODO(), foundSts)
-		if err != nil {
-			return err
+		foundSTSSize := *foundSts.Spec.Replicas
+		newSTSSize := *sts.Spec.Replicas
+		if newSTSSize < foundSTSSize {
```

Comment:
What about scale up events? Don't we need to handle them here?

Reply:
No Scale up does not need any special handling.